### PR TITLE
Dynamically fetch websocket subscription url

### PR DESCRIPTION
**Pull request by toini**

### CODE REVIEW

**Comment by leetal** [comment]
This is of utter importance at the moment since Tibber decided to enable the new WSS enpoints ahead of its planned December launch 🤦‍♂️The Home Assistant integration (+ any more that relies on this library) will be unavailable unless manually edited.

**Comment by toini** [comment]
@leetal the new websocket server is available for testing. Note that there was an unrelated outage in api.tibber.com (affecting websockets) yesterday between 13 - 21 CET.

File: tibber/__init__.py [hunk 1]
```diff
@@ -196,7 +201,7 @@ async def update_info(self) -> None:
             self._all_home_ids += [home_id]
             if not (subs := _home.get("subscriptions")):
                 continue
-            if subs[0].get("status", "ended").lower() == "running":
+            if subs[0].get("status", "ended") is not None and subs[0].get("status", "ended").lower() == "running":
```
Comment (toini) [inline tibber/__init__.py:204]: This is not related to fetching websocket url but i was getting errors when testing on a home with no subs

**Review by Danielhiversen — CHANGES_REQUESTED** [review]

File: tibber/__init__.py [hunk 2]
```diff
@@ -65,14 +64,9 @@ def __init__(
...
         self._all_home_ids: list[str] = []
         self._homes: dict[str, TibberHome] = {}
-        self.sub_manager: Client = Client(
-            transport=TibberWebsocketsTransport(
-                SUB_ENDPOINT,
-                self._access_token,
-                self.user_agent,
-            ),
-        )
+        self.sub_manager: Client = None  # type: ignore
```
[the remark this answers is not in the document]
Comment (toini) [reply tibber/__init__.py]: I think this made mypy fail now

[the remark this answers is not in the document]
Comment (toini) [reply tibber/__init__.py]: Did this appear after the change?

```Run mypy --disallow-incomplete-defs --no-implicit-optional $SRC_FOLDER
tibber/__init__.py:82: error: Item "None" of "Optional[Client]" has no attribute "close_async"  [union-attr]
tibber/__init__.py:89: error: Argument 1 to "TibberWebsocketsTransport" has incompatible type "None"; expected "str"  [arg-type]
tibber/__init__.py:281: error: Item "None" of "Optional[Client]" has no attribute "transport"  [union-attr]
tibber/__init__.py:282: error: Item "None" of "Optional[Client]" has no attribute "transport"  [union-attr]
Found 4 errors in 1 file (checked 4 source files)
Error: Process completed with exit code 1.
```

[the remark this answers is not in the document]
Comment (Danielhiversen) [reply tibber/__init__.py]: With your changes self.sub_manager can now be None.
So you need to test if self.sub_manager is None before using it.
For example In [rt_disconnect](https://github.com/Danielhiversen/pyTibber/blob/0.26.7/tibber/__init__.py#L82)

**Comment by p4fg** [comment]
Any progress on this? Waiting with anticipation!

**Comment by toini** [comment]
@p4fg you can clone the latest code. It works.

**Comment by p4fg** [comment]
Yeah, im using it via pip3 in a build-pipeline... so i'll see what i can do.

**Comment by Danielhiversen** [comment]
[0.27.0 ](https://github.com/Danielhiversen/pyTibber/releases/tag/0.27.0) is now released.

Home Assistant would also need to be updated to use the latest library and the new endpoint.

**Comment by toini** [comment]
@Danielhiversen like this? https://github.com/tibber/core/commit/5cb70ced1ad72c389e45b69ce3037fc885064535

**Pull request merged**